Removed Legacy (pre-1.9.3) support.

Seberius opened the pull request:
Hello,
As discussed in #7, this branch removes the legacy (pre-1.9.3) cache and cleans up some Ruby 1.8 support code.
#### Tests

This branch is currently passing all tests on 2.0.0-p643, 2.2.0, jruby-1.7.19 and jruby-9.0.0.0-pre1.

```
Fabulous run in 0.002440s, 9426.2295 runs/s, 27459.0164 assertions/s.

23 runs, 67 assertions, 0 failures, 0 errors, 0 skips
```

Commit:
Removed legacy cache implementation and cleaned up legacy support.

SamSaffron commented on lib/lru_redux.rb:
```diff
@@ -1,18 +1,8 @@
-  require "lru_redux/cache"
-end
+require "lru_redux/cache"
+require "lru_redux/cache_legacy" if RUBY_VERSION < "2.1.0"
```
should we check for mri explicitly here?

SamSaffron commented:
Let's make sure there is a changelog message here explaining what just happened and what people on 1.8 need to do, also bump the version up now in master so its clearer

Seberius commented:
Good catch on the legacy check.  We had discussed a major version bump, so would it be v1.0.0?  I currently have it marked as version NEXT in the changelog.  Also, to clarify, you want to do the version bump on this merge, before the TTL cache branch?

SamSaffron commented:
we should probably combine, but definitely want a version bump marked in
source control here.

v1.0.0 sounds fine to me for this ... maybe we release this and then do 1.1
for ttl cache shortly after?

On Fri, Mar 27, 2015 at 1:19 PM, Kaijah Hougham <notifications@github.com>
wrote:

> Good catch on the legacy check. We had discussed a major version bump, so
> would it be v1.0.0? I currently have it marked as version NEXT in the
> changelog. Also, to clarify, you want to do the version bump on this merge,
> before the TTL cache branch?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SamSaffron/lru_redux/pull/8#issuecomment-86789593>.
>

Commit:
Version bump.
Legacy check now checks for MRI.
Readme updated with instructions for Ruby 1.8 users.
Changelog update and typo fix.
Gemspec now specifies 1.9.3 as the minimum Ruby version.
GitHub address added to gemspec.

Seberius commented:
I marked this branch with 1.0.0 in the changelog and version.rb, along with an addition to the install instructions about Ruby 1.8.  Version 1.1 for TTL sounds good.  I also added the repo address to the gemspec to make getting here a little easier.

SamSaffron commented on lru_redux.gemspec:
```diff
@@ -10,9 +10,11 @@ Gem::Specification.new do |spec|
   spec.email         = ["sam.saffron@gmail.com"]
```
you can add yourself to authors / emails here, at this point you have added plenty to deserve that :)

Seberius commented:
> you can add yourself to authors / emails here, at this point you have added plenty to deserve that :)

Thank you!

stanio commented on lib/lru_redux/cache.rb:
```diff
@@ -1,104 +1,75 @@
-    a = to_a
-    a.each do |pair|
+    array = @data.to_a
+    array.reverse!.each do |pair|
```
Could this use [`Hash#reverse_each`](https://ruby-doc.org/core-2.3.4/Enumerable.html#method-i-reverse_each) instead?

```ruby
  def each(*args, &block)
    @data.reverse_each(*args, &block)
  end
```
